src/sparseml/exporters/transforms/fold_identity_initializers.py

```diff
@@ -0,0 +1,69 @@
+# Copyright (c) 2021 - present / Neuralmagic, Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import logging
+
+from onnx import ModelProto
+
+from sparseml.exporters.transforms import OnnxTransform
+from sparseml.exporters.transforms.utils import get_structural_matches
+from sparseml.onnx.utils import ONNXGraph
+
+
+__all__ = ["FoldIdentityInitializers"]
+
+_LOGGER = logging.getLogger(__name__)
+
+
+def fold_identity_initializer(
+    match: "MatchResult", model: ModelProto
+) -> ModelProto:  # noqa F821
+    """
+    Find any node in the graph that uses the output of
+    the `match.node` as it's input. Replace the input
+    of this node with `match.node`'s input. Finally,
+    remove `match.node` from the graph.
+
+    :param match: Match node to be folded into the graph
+    :param model: ONNX model to be transformed
+    :return: ONNX model with the match node folded into the graph
+    """
+    graph = ONNXGraph(model)
+    for child_node in graph.get_node_children(match.node):
+        for i, child_node_input in enumerate(child_node.input):
+            if child_node_input == match.node.output[0]:
+                child_node.input[i] = match.node.input[0]
+        model.graph.node.remove(match.node)
+    return model
+
+
+class FoldIdentityInitializers(OnnxTransform):
+    """
+    Folds any `Identity` initializer node into the graph.
+    TODO: Add graph
+    """
+
+    def transform(self, model: ModelProto) -> ModelProto:
+        count_converted_nodes = 0
+        graph = ONNXGraph(model)
+        for match in get_structural_matches(
```

Comment:
I am wondering whether this match check is strong enough. Originally (in the pre-refactor code), this was the set of criteria needed for the check to pass:

```python
  def is_match(node: onnx.NodeProto) -> bool:
        return (
            node.op_type == "Identity"
            and len(node.input) == 1
            and len(node.output) == 1
            and node.input[0] in graph._name_to_initializer
        ]
```
But from my quick study of identity ONNX OP (http://www.xavierdupre.fr/app/mlprodict/helpsphinx/onnxops/onnx__Identity.html) it seems that all those need to be also fulfilled to create a valid identity op in the onnx framework, so we may not need to explicitly encode them here

Reply:
🔥 nice!!

Reply:
Ehh I guess not, it's pretty clear from the onnx docs that this is the case https://github.com/onnx/onnx/blob/main/docs/Operators.md#Identity